tests/test_llm_json_parser.py

```diff
@@ -0,0 +1,142 @@
+import textwrap
+
+import pytest
+
+from paperqa.core import llm_parse_json
+
+
+# test cases obtained from test_paperqa.py
+@pytest.mark.parametrize(
+    "example",
+    [
+        """Sure here is the json you asked for!
+
+    {
+    "example": "json"
+    }
+
+    Did you like it?""",
+        '{"example": "json"}',
+        """
+```json
+{
+    "example": "json"
+}
+```
+
+I have written the json you asked for.""",
+        """
+
+{
+    "example": "json"
+}
+
+""",
+    ],
+)
+def test_llm_parse_json(example: str) -> None:
+    assert llm_parse_json(example) == {"example": "json"}
+
+
+def test_llm_parse_json_newlines() -> None:
+    """Make sure that newlines in json are preserved and escaped."""
+    example = textwrap.dedent(
+        """
+        {
+        "summary": "A line
+
+        Another line",
+        "relevance_score": 7
+        }"""
+    )
+    assert llm_parse_json(example) == {
+        "summary": "A line\n\nAnother line",
+        "relevance_score": 7,
+    }
+
+
+# Additional cases
+# Case 1: Removing Think Tags
```

Comment:
Can you move this case descriptor to `pytest.param(..., id="removing-think-cases")`